CONTRIBUTING.md

```diff
@@ -16,13 +16,36 @@ Lastly, run the test command from the console:
 
     npm test
 
+
 ## Deploying
 
-- `npm run prepublish && npm run docs`
-- Update the version key in [package.json](https://github.com/mapbox/mapbox-gl-geocoder/blob/master/package.json#L3)
-- Outline changes in [CHANGELOG.md](https://github.com/mapbox/mapbox-gl-geocoder/blob/master/CHANGELOG.md)
-- Commit and push
-- `git tag -a vX.X.X -m 'vX.X.X'`
-- `git push --tags`
-- `npm publish`
-- Update version number in GL JS examples ([one](https://github.com/mapbox/mapbox-gl-js/blob/publisher-production/docs/pages/example/mapbox-gl-geocoder.html), [two](https://github.com/mapbox/mapbox-gl-js/blob/publisher-production/docs/pages/example/point-from-geocoder-result.html), [three](https://github.com/mapbox/mapbox-gl-js/blob/publisher-production/docs/pages/example/mapbox-gl-geocoder-outside-the-map.html), [four](https://github.com/mapbox/mapbox-gl-js/blob/publisher-production/docs/pages/example/mapbox-gl-geocoder-limit-region.html), [five](https://github.com/mapbox/mapbox-gl-js/blob/publisher-production/docs/pages/example/mapbox-gl-geocoder-accept-coordinates.html), [six](https://github.com/mapbox/mapbox-gl-js/blob/publisher-production/docs/pages/example/mapbox-gl-geocoder-proximity-bias.html), [seven](https://github.com/mapbox/mapbox-gl-js/blob/publisher-production/docs/pages/example/forward-geocode-custom-data.html))
+Follow this deploy process after all changes for the release are merged into master. You will copy and paste this checklist in the comment of the release pull request.
+
+```
+## Release checklist
+
+1. Create a branch off `master` and a pull request with the following changes. Copy this checklist in the comment of the pull request.
+    - [ ] Update the [CHANGELOG.md](https://github.com/mapbox/mapbox-gl-geocoder/blob/master/CHANGELOG.md) by comparing the last release and what is on `master`. In the changelog, replace the `master` heading with the to-be-released stable version.
+    - [ ] Update the version number in `package.json` and `package.lock.json`.
+3. Request a review PR and then merge it into `master`.
+4. Tag the release and start the build.
+    - [ ] Make sure you've pulled in all changes to `master` locally.
+    - [ ] Build the release with `npm run prepublish && npm run docs`
+    - [ ] Commit and push with commit message `vX.X.X`
+    - [ ] Create the git tag for the release with `git tag -a vX.X.X -m 'vX.X.X'`
+    - [ ] Push the tags with `git push --tags`
+    - [ ] Run `npm publish`
```

Comment:
I've been bitten in the past with non-committed files containing secret keys in the same directory, which to much horror `npm publish` silently packaged up and included in the deploy :scream:. In that regard I find it safer to use CI to publish the release, that said so long as people doing the npm publish are aware of this then it should be okay.

Reply:
> find it safer to use CI to publish the release

Oh, smart! I'm not familiar with this, what does that look like in practice?